public/app/features/alerting/unified/components/rule-list/extensions/EnrichmentDrawerExtension.tsx

```diff
@@ -0,0 +1,42 @@
+import { ComponentType, memo } from 'react';
+
+import { t } from '@grafana/i18n';
+import { withErrorBoundary } from '@grafana/ui';
+
+import { logError } from '../../../Analytics';
+
+export interface EnrichmentDrawerExtensionProps {
+  ruleUid: string;
+  onClose: () => void;
+}
+
+// Internal variable to store the extension component, for now only one component is supported
+let InternalEnrichmentDrawerExtension: ComponentType<EnrichmentDrawerExtensionProps> | null = null;
+
+// This component is used to render the enrichment drawer extension.
+const EnrichmentDrawerExtensionComponent: ComponentType<EnrichmentDrawerExtensionProps> = (props) => {
+  if (!InternalEnrichmentDrawerExtension) {
+    return null;
+  }
+
+  // Wrap the component with error boundary
+  const WrappedComponent = withErrorBoundary(InternalEnrichmentDrawerExtension, {
+    title: t(
+      'alerting.enrichment.error-boundary.enrichment-drawer-extension',
+      'Enrichment Drawer Extension failed to load'
+    ),
+    style: 'alertbox',
+    errorLogger: logError,
+  });
+
+  return <WrappedComponent {...props} />;
+};
+
+export const EnrichmentDrawerExtension = memo(EnrichmentDrawerExtensionComponent, (prevProps, nextProps) => {
```

Comment:
I'm quite concerned about the way we include enterprise features into the OSS code.

I expressed my thoughts earlier when working on AI extensions https://github.com/grafana/grafana/pull/107754#discussion_r2197665804
We planned to tackle OSS - Enterprise integration in a follow up PRs, but sadly that never happened and this integration patter has become more common in our codebase

I feel it's the high time to ensure this is the way we want to integrate enterprise moving forward or if we should stop here and discuss what other options we have

Reply:
ok, you're right. Let me think about it.